code/__HELPERS/game.dm

```diff
@@ -269,48 +255,31 @@
 
 	return found_mobs
 
-// Returns a list of hearers in view(R) from source (ignoring luminosity). Used in saycode. Consider using `hearers` when you *only* need mobs.
-/proc/get_hearers_in_view(R, atom/source)
-	var/turf/T = get_turf(source)
+/// Returns a list of hearers in view(view_radius) from source (ignoring luminosity). uses important_recursive_contents[RECURSIVE_CONTENTS_HEARING_SENSITIVE]
+/proc/get_hearers_in_view(view_radius, atom/source)
+	var/turf/center_turf = get_turf(source)
 	. = list()
-
-	if(!T)
+	if(!center_turf)
 		return
-
-	var/list/processing_list = list()
-	if (R == 0) // if the range is zero, we know exactly where to look for, we can skip view
-		processing_list += T.contents // We can shave off one iteration by assuming turfs cannot hear
-	else  // A variation(?) of get_hear inlined here to take advantage of the compiler's fastpath for obj in view
-		// Benchmark before trying to "optimize" this proc
-		processing_list += hearers(R, T)
-
-		// dview is inlined here
-		GLOB.dview_mob.loc = T
-		GLOB.dview_mob.see_invisible = SEE_INVISIBLE_MINIMUM // Sorry, invisible objects. You folks can't HEAR_1 anyway.
-		for(var/obj/O in view(R, GLOB.dview_mob))
-			processing_list += O
-		GLOB.dview_mob.loc = null
-
-	while(length(processing_list)) // recursive_hear_check inlined here with optimized processing
-		var/atom/A = processing_list[length(processing_list)]
-		if(A.flags_1 & HEAR_1)
-			. += A
-		processing_list.len--
-		processing_list += A.contents
+	var/lum = center_turf.luminosity
+	center_turf.luminosity = 6 // This is the maximum luminosity
+	for(var/atom/movable/movable in view(view_radius, center_turf))
+		var/list/recursive_contents = LAZYACCESS(movable.important_recursive_contents, RECURSIVE_CONTENTS_HEARING_SENSITIVE)
+		if(recursive_contents)
+			. += recursive_contents
+	center_turf.luminosity = lum
 
 /proc/get_mobs_in_radio_ranges(list/obj/item/radio/radios)
 	. = list()
 	// Returns a list of mobs who can hear any of the radios given in @radios
 	for(var/obj/item/radio/R in radios)
```

Comment:
Out of curiosity can this be made typeless?

Reply:
not sure if things like lighting overlays and turfs should hear. and the `important_recursive_contents` var is only on atom/movable